server/backend/api/app/routers/router_user_role_assignment.py

```diff
@@ -67,20 +67,25 @@ def create_user_role_assignment(
     )
 
     try:
-
-        requesting_user = get_requesting_user(db, requester.cognito_user_id)
         role = crud_role.get_role(db, role_assignment_request.role_id)
 
         audit_event_log.role = role
         audit_event_log.application = role.application
-        audit_event_log.requesting_user = requesting_user
+        audit_event_log.requesting_user = requester
 
-        return crud_user_role.create_user_role(
+        new_user_role_assignment =  crud_user_role.create_user_role(
             db,
             role_assignment_request,
             target_user,
-            requesting_user.cognito_user_id
+            requester.cognito_user_id
+        )
+        # get target user from database, so for existing user, we'll have get the cognito user id
+        audit_event_log.target_user = crud_user.get_user_by_domain_and_guid(
```

Comment:
instead of get user by domain and user name, we get user by domain and user guid now.

Reply:
Just wondering, the "target_user" and the crud_user.get_user_by_domain_and_guid() are having different types, does it work for variable assignment?